src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -116,94 +116,82 @@ open class Singleton<T : Entity<T>> constructor(val entityCtor: () -> T) : Handl
         entity = entityCtor.invoke().decodeEntity(added)
     }
 
-    fun get(): T? {
-        return entity
-    }
+    fun get() = entity
```

Comment:
Rather than having `get` and `set` methods, consider making `entity` a public property with a custom setter.

```kotlin
var entity: T? = null
  set(value) {
    requireNotNull(value)
    field = value
    singletonSet(
      particle.toWasmAddress(),
      toWasmAddress(),
      value.encodeEntity().toWasmString()
    )
  }
```

Reply:
get() and set() are the same api as you get in other languages (JS, C++) and I think we would like it to keep it this way. I'll leave it as is for now at least.

Reply:
Hmm. Not sure I agree... I think the apis should be idiomatic-ish with for the language they're intended for.

Reply:
Normally I'd agree, however Singleton/Collection/Reference are in a type hierarchy with Handle. They are all handles, and all of them need to use methods to manipulate. Making Singleton use setters/getters would make it the odd one out, unless we add operators to manipulate Collections and References too.

Also, there's a weird issue with nulls right now. The way you null out a Singleton is via Singleton.clear(), not by Singleton.set(null)